dist/client-integration.js

```diff
@@ -0,0 +1,454 @@
+      secureConnection: false,
+      storage: {
+        INBOX: {
+          messages: [{
+            raw: 'Subject: hello 1\r\n\r\nWorld 1!'
+          }, {
+            raw: 'Subject: hello 2\r\n\r\nWorld 2!',
+            flags: ['\\Seen']
+          }, {
+            raw: 'Subject: hello 3\r\n\r\nWorld 3!',
+            uid: 555
+          }, {
+            raw: 'From: sender name <sender@example.com>\r\nTo: Receiver name <receiver@example.com>\r\nSubject: hello 4\r\nMessage-Id: <abcde>\r\nDate: Fri, 13 Sep 2013 15:01:00 +0300\r\n\r\nWorld 4!'
+          }, {
+            raw: 'Subject: hello 5\r\n\r\nWorld 5!',
+            flags: ['$MyFlag', '\\Deleted'],
+            uid: 557
+          }, {
+            raw: 'Subject: hello 6\r\n\r\nWorld 6!'
+          }, {
+            raw: 'Subject: hello 7\r\n\r\nWorld 7!',
+            uid: 600
+          }]
+        },
+        '': {
+          separator: '/',
+          folders: {
+            '[Gmail]': {
+              flags: ['\\Noselect'],
+              folders: {
+                'All Mail': {
+                  'special-use': '\\All'
+                },
+                Drafts: {
+                  'special-use': '\\Drafts'
+                },
+                Important: {
+                  'special-use': '\\Important'
+                },
+                'Sent Mail': {
+                  'special-use': '\\Sent'
+                },
+                Spam: {
+                  'special-use': '\\Junk'
+                },
+                Starred: {
+                  'special-use': '\\Flagged'
+                },
+                Trash: {
+                  'special-use': '\\Trash'
+                },
+                A: {
+                  messages: [{}]
+                },
+                B: {
+                  messages: [{}]
+                }
+              }
+            }
+          }
+        }
+      }
+    };
+    server = (0, _hoodiecrowImap.default)(options);
+    server.listen(port, done);
+  });
+  afterEach(done => {
+    server.close(done);
+  });
+  describe('Connection tests', () => {
+    var insecureServer;
+    beforeEach(done => {
+      // start imap test server
+      var options = {
+        // debug: true,
+        plugins: [],
+        secureConnection: false
+      };
+      insecureServer = (0, _hoodiecrowImap.default)(options);
+      insecureServer.listen(port + 2, done);
+    });
+    afterEach(done => {
+      insecureServer.close(done);
+    });
+    it('should use STARTTLS by default', () => {
+      imap = new _.default('127.0.0.1', port, {
+        logLevel: _.LOG_LEVEL_NONE,
+        auth: {
+          user: 'testuser',
+          pass: 'testpass'
+        },
+        useSecureTransport: false
+      });
+      return imap.connect().then(() => {
+        expect(imap.client.secureMode).to.be.true;
+      }).then(() => {
+        return imap.close();
+      });
+    });
+    it('should ignore STARTTLS', () => {
```

Comment:
this doesn't look right.

1. I would make the lib export dist only to npm, by default folder should be ignored by git
2. this build includes tests? definitely fishy

Reply:
ok, it seems its a little too late to ignore dist folder
but for build you should run build script https://github.com/emailjs/emailjs-imap-client/blob/master/scripts/build.sh which ingores `-unit` and `-ingegration` files